client/Components/stockCard.jsx

```diff
@@ -1,32 +1,85 @@
 'use strict';
-
+//NB.this page need to have the this.props.inputID to work
 import React from 'react';
 import { render } from 'react-dom';
 import { connect } from 'react-redux';
 import { toJS } from 'immutable';
 
+const Card = require('material-ui/lib/card/card');
+const CardActions = require('material-ui/lib/card/card-actions');
+const CardExpandable = require('material-ui/lib/card/card-expandable');
+const CardHeader = require('material-ui/lib/card/card-header');
+const CardMedia = require('material-ui/lib/card/card-media');
+const CardText = require('material-ui/lib/card/card-text');
+const CardTitle = require('material-ui/lib/card/card-title');
+const RaisedButton = require('material-ui/lib/raised-button');
+
 export const Stock = React.createClass({
 
+  //place to hold temporary variables
+  getInitialState() {
+      return {
+        numSharesToSell: 0
+      };
+  },
+
   render() {
-    const { sell, stockSymbol, shares, matchID, userID, price } = this.props;
-    console.log('matchID is ' + matchID + ' in Stock');
+    var that = this;
+    const { sell, stockSymbol, shares, matchID, userID, price, bid, ask, percent_change, gain_loss, marketValue, name } = this.props;
+    
     return (
-      <div>
-        <p> {stockSymbol} : {shares} at ${price} </p>
-        <input type="number" id={this.props.inputID} min="1" max={shares} step="1" />
-        <button onClick={() => {
-          let numSharesToSell = Number(document.getElementById(this.props.inputID).value);
-          let sellOptions = {
-            numShares: numSharesToSell,
-            stockTicker: stockSymbol,
-            matchID: matchID,
-            userID: userID,
-            action: 'sell',
-            // hardcode price until AJAX call works
-            // price: '112'
-          };
-          sell(sellOptions); // trigger action creator in actions.js
-        }}> Sell </button>
+      <div className="container">
+
+        <Card>
+          <div className="row container">
+            <div className="three columns">
+              <h4>{name}<span>{stockSymbol}</span></h4>
+            </div>
+
+            <div className="two columns">
+              <h5> Shares: {shares}</h5>
+            </div>
+
+            <div className="two columns">
+              <h5> Total: ${(shares * price).toFixed(2)}</h5>
+            </div>
+
+            <div className="three columns">
+              <input type="number" ref="sellNum" min="1" max={shares} step="1" onChange={function (event) {
+                that.setState({numSharesToSell: event.target.value})
+              }} />
```

Comment:
again, setState is against redux flow. Maybe a solution would be setting a key on the element to numSharesToSell (i.e. this.numSharesToSell = 0)? It works if we use es6 fat-arrow notation, because the callback won't lose its context. See joinMatch.jsx for an example of setting properties on the object itself (self.matches)